### PR TITLE
Fixing destroy when role scope is a Management Group

### LINKED ISSUE #5125: terraform destroy fails for azurerm_role_definition if role scope is a Management Group
When using **azurerm_role_definition** to create a role scoped to a Management  Group instead to a Subscription or Resource Group, `terraform destroy` fails cause the scope sent to the API is empty.

This seems to be casued by the *parseRoleDefinitionId* function defined [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.38.0/azurerm/resource_arm_role_definition.go), which asumes that every roleDefinitionId starts with the scope which is not the case for Roles scoped to a Management Group.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version 

Terraform v0.12.17
+ provider.azurerm v1.38.0

### Affected Resource(s)

* `azurerm_role_definition`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = ">= 1.38.0"
  }
}

variable "mg_id" {
  type        = string
  description = "Existing Management Group ID"
}

locals {
  scope = "/providers/Microsoft.Management/managementGroups/${var.mg_id}"
}

resource "azurerm_role_definition" "RoleInManagementGroup" {
  name        = "[REMOVE] Role In Management Group"
  scope       = local.scope
  description = "This is a custom role created via Terraform"

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [
    local.scope
  ]
}
```

### Debug Output

https://gist.github.com/cmendible/e44f10b665ba9c97cb9a9d1167c5bd99#file-debug-log

### Expected Behavior

Role must be deleted by `terraform destroy`

### Actual Behavior

`terraform destroy` results in:

Error deleting Role Definition "00000000-0000-0000-0000-000000000000" at Scope "": authorization.RoleDefinitionsClient#Delete: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'live.com#terraformuser@outlook.com' with object id '00000000-0000-0000-0000-000000000000' does not have authorization to perform action 'Microsoft.Authorization/roleDefinitions/delete' over scope '/providers/Microsoft.Authorization/roleDefinitions/fc4a1caf-aee2-b29b-14e4-11de9aa09cd6' or the scope is invalid. If access was recently granted, please refresh your credentials."

Issue seems to be related to the fact that Terraform is sending en empty scope.

### Steps to Reproduce

You'll need to have an existing management group. The config file will ask for the ID of the group.

1. `terraform apply`
2. `terraform destroy`

### PR DESCRIPTION
**Pull request by cmendible**
Fixes #5125: When using **azurerm_role_definition** to create a role scoped to a Management  Group instead to a Subscription or Resource Group, `terraform destroy` failed cause the scope sent to the API was empty.

This was caused by the *parseRoleDefinitionId* function which asumes that every roleDefinitionId starts with the scope which is not the case for Roles scoped to a Management Group.

### CODE REVIEW

**Comment by cmendible** [comment]
Hi @katbyte no problem I'll add the tests and come back to you with sample IDs

**Comment by cmendible** [comment]
@katbyte  when roles are scoped to a management group, the ID takes the following form:

`/providers/Microsoft.Authorization/roleDefinitions/<GUID>`  

so the scope can not be extracted from it and therefore we must be read the _assignable_scopes_ property.

File: azurerm/internal/services/authorization/tests/resource_arm_role_definition_test.go [hunk 1]
```diff
@@ -170,6 +170,26 @@ func TestAccAzureRMRoleDefinition_emptyName(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMRoleDefinition_managementGroup(t *testing.T) {
```
Comment (cmendible) [inline azurerm/internal/services/authorization/tests/resource_arm_role_definition_test.go]: @katbyte as per my understanding this should test the changes, if TF_ACC is set.

**Comment by cmendible** [comment]
> Error checking for presence of

Sorry about that, the management group must exist before the test runs. Is there a place where we can create this dependency before running the tests or should I embed the management group creation inside the `testAccAzureRMRoleDefinition_managementGroup`  method?

**Comment by cmendible** [comment]
@katbyte just added the management group to the test.

**Comment by cmendible** [comment]
> @katbyte just added the management group to the test.

@katbyte can you force and e2e test?

**Comment by cmendible** [comment]
Hi @katbyte any news on this one?

**Comment by javierjeronimo** [comment]
Big organisations use Management-Groups to manage Azure Policy and Custom-Roles definitions, this is really needed. Thanks.

**Comment by chirurico** [comment]
I have the same problem. It would really be helpful for Azure-Terraform interactions regarding resource groups.

File: azurerm/internal/services/authorization/role_definition_resource.go [hunk 2]
```diff
@@ -217,7 +222,9 @@ func resourceArmRoleDefinitionDelete(d *schema.ResourceData, meta interface{}) e
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parseRoleDefinitionId(d.Id())
+	scope := d.Get("scope").(string)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/authorization/role_definition_resource.go:225]: we can't retrieve fields from the state during the read (there's no guarantees they're present) - so we'll need to pull this from somewhere else (e.g. look it up / put it in the ID?)

Comment (cmendible) [reply azurerm/internal/services/authorization/role_definition_resource.go]: this was fixed by @jackofallops in `azurerm/internal/services/authorization/parse/role_definition.go` so the scope is saved in the ID

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by jackofallops** [comment]
Tests passing, (failures unrelated)
image

**Pull request merged**